smoke-test/src/test/java/org/opennms/smoketest/DaemonConfigPageIT.java

```diff
@@ -0,0 +1,258 @@
+                final String reloadStateText = getDaemonRowElement().findElements(By.xpath("./td[3]/div/span")).get(0).getText();
+                return (reloadStateText.equals("Success") || reloadStateText.equals("Failed") || reloadStateText.equals("Unknown"));
+            });
+
+            //Check Ui State after the Reload terminated and return the reloadStateText
+            reloadCell = getDaemonRowElement().findElement(By.xpath("./td[3]"));
+            reloadButton = reloadCell.findElement(By.xpath("./div/button"));
+            Assert.assertThat(reloadButton.isDisplayed(), is(true));
+            Assert.assertThat(reloadButton.isEnabled(), is(true));
+
+            reloadCellStateSpan = reloadCell.findElements(By.xpath("./div/span")).get(0);
+            Assert.assertThat(reloadCellStateSpan.isDisplayed(), is(true));
+
+            return reloadCellStateSpan.getText();
+        }
+
+        public String getCurlString() {
+            if (!this.isReloadable()) {
+                throw new IllegalStateException("This daemon is not reloadable");
+            }
+
+            WebElement preElement = getPopupPreElement();
+            Assert.assertThat(preElement.isDisplayed(), is(false));
+            Assert.assertThat(preElement.isEnabled(), is(true));
+
+            WebElement curlButton = getDaemonRowElement().findElement(By.xpath("./td[3]/div/span/button"));
+            Assert.assertThat(curlButton.isDisplayed(), is(true));
+            Assert.assertThat(curlButton.isEnabled(), is(true));
+            curlButton.click();
+
+            preElement = getPopupPreElement();
+            Assert.assertThat(preElement.isDisplayed(), is(true));
+            Assert.assertThat(preElement.isEnabled(), is(true));
+
+            final String curlString = preElement.getText();
+
+            curlButton = getDaemonRowElement().findElement(By.xpath("./td[3]/div/span/button"));
+            Assert.assertThat(curlButton.isDisplayed(), is(true));
+            Assert.assertThat(curlButton.isEnabled(), is(true));
+            curlButton.click();
+
+            preElement = getPopupPreElement();
+            Assert.assertThat(preElement.isDisplayed(), is(false));
+            Assert.assertThat(preElement.isEnabled(), is(true));
+
+            return curlString;
+        }
+
+        private WebElement getDaemonRowElement() {
+            return this.page.getRowForDaemonName(this.daemonName);
+        }
+
+        private WebElement getPopupPreElement() {
+            return this.page.getPopupPreElementForDaemonName(this.daemonName);
+        }
+    }
+
+    private class DaemonReloadPage {
+
+        private final int expectedDaemonCount = 30;
+
+        public DaemonReloadPage open() {
+            m_driver.get(baseURI + "opennms/admin/daemons/index.jsp");
+            new WebDriverWait(m_driver, 5).until((Predicate<WebDriver>) (driver) -> getDaemonRows().size() == expectedDaemonCount);
```

Comment:
This would probably be better manually verified when invoking the open() instead of here.
Otherwise `expectedDaemonCount` should be a parameter, which makes the API strange to use.